src/specific/s_input.c

```diff
@@ -549,35 +890,124 @@ INPUT_STATE S_Input_GetCurrentState(INPUT_LAYOUT layout_num)
+                break;
+            case INPUT_ROLE_ROLL:
+                linput.roll                   |= btn_state;
+                break;
+            case INPUT_ROLE_OPTION:
+                linput.option                 |= btn_state;
+                // TODO Should deselect always be linked to option?
+                linput.deselect               |= btn_state;
+                break;
+            case INPUT_ROLE_FLY_CHEAT:
+                linput.fly_cheat              |= btn_state;
+                break;
+            case INPUT_ROLE_ITEM_CHEAT:
+                linput.item_cheat             |= btn_state;
+                break;
+            case INPUT_ROLE_LEVEL_SKIP_CHEAT:
+                linput.level_skip_cheat       |= btn_state;
+                break;
+            case INPUT_ROLE_TURBO_CHEAT:
+                linput.turbo_cheat            |= btn_state;
+                break;
+            case INPUT_ROLE_PAUSE:
+                linput.pause                  |= btn_state;
+                break;
+            case INPUT_ROLE_CAMERA_UP:
+                linput.camera_up              |= btn_state;
+                break;
+            case INPUT_ROLE_CAMERA_DOWN:
+                linput.camera_down            |= btn_state;
+                break;
+            case INPUT_ROLE_CAMERA_LEFT:
+                linput.camera_left            |= btn_state;
+                break;
+            case INPUT_ROLE_CAMERA_RIGHT:
+                linput.camera_right           |= btn_state;
+                break;
+            case INPUT_ROLE_CAMERA_RESET:
+                linput.camera_reset           |= btn_state;
+                break;
+            case INPUT_ROLE_EQUIP_PISTOLS:
+                linput.equip_pistols          |= btn_state;
+                break;
+            case INPUT_ROLE_EQUIP_SHOTGUN:
+                linput.equip_shotgun          |= btn_state;
+                break;
+            case INPUT_ROLE_EQUIP_MAGNUMS:
+                linput.equip_magnums          |= btn_state;
+                break;
+            case INPUT_ROLE_EQUIP_UZIS:
+                linput.equip_uzis             |= btn_state;
+                break;
+            case INPUT_ROLE_USE_SMALL_MEDI:
+                linput.use_small_medi         |= btn_state;
+                break;
+            case INPUT_ROLE_USE_BIG_MEDI:
+                linput.use_big_medi           |= btn_state;
+                break;
+            case INPUT_ROLE_SAVE:
+                linput.save                   |= btn_state;
+                break;
+            case INPUT_ROLE_LOAD:
+                linput.load                   |= btn_state;
+                break;
+            case INPUT_ROLE_FPS:
+                linput.toggle_fps_counter     |= btn_state;
+                break;
+            case INPUT_ROLE_BILINEAR:
+                linput.toggle_bilinear_filter |= btn_state;
+                break;
+            default:
+                break;
```

Comment:
Can we turn this into a function?
`static INPUT_STATE S_Input_Toggle(INPUT_STATE state, INPUT_ROLE role, bool state);`